### PR TITLE
Migrate-web3-to-ether.js

### PR DESCRIPTION
**Pull request by adetoye-dev**

### CODE REVIEW

File: src/components/Post/GovernanceSideBar/Proposals/SecondProposalEth.tsx [hunk 1]
```diff
@@ -235,14 +234,10 @@ const SecondProposalEth = ({ className, proposalId, seconds }: SecondProposalPro
 			}
 		}
 
-		const voteContract = new web3.eth.Contract(abi, contractAddress);
+		const voteContract = new Contract(contractAddress, abi, await web3.getSigner());
 
-		voteContract.methods
+		voteContract
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/Proposals/SecondProposalEth.tsx:239]: please add an `await` here

File: src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEth.tsx [hunk 2]
```diff
@@ -294,17 +295,13 @@ const VoteReferendum = ({ className, referendumId, onAccountChange, lastVote, se
...
 		setLoadingStatus({ isLoading: true, message: 'Awaiting block confirmation' });
 
-		const voteContract = new web3.eth.Contract(abi, contractAddress);
+		const voteContract = new Contract(contractAddress, abi, await web3.getSigner());
 
 		// estimate gas.
 		//https://docs.moonbeam.network/builders/interact/eth-libraries/deploy-contract/#interacting-with-the-contract-send-methods
 
-		voteContract.methods
+		voteContract
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEth.tsx:303]: add an `await` here please

File: next.config.js
Comment (alphainfinitus) [inline next.config.js:1]: Remove this from Prod please

File: src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEthV2.tsx [hunk 3]
```diff
@@ -616,7 +616,8 @@ const VoteReferendumEthV2 = ({ className, referendumId, onAccountChange, lastVot
 								onBalanceChange={(balance: BN) => setLockedBalance(balance)}
 								convictionClassName={className}
 								handleSubmit={async () => await handleSubmit()}
-								disabled={!wallet || !lockedBalance || isBalanceErr || lockedBalance.lte(ZERO_BN)}
+								disabled={false}
+								// disabled={!wallet || !lockedBalance || isBalanceErr || lockedBalance.lte(ZERO_BN)}
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/Referenda/VoteReferendumEthV2.tsx:620]: please add the correct disabled state conditions

File: src/components/Post/GovernanceSideBar/index.tsx [hunk 4]
```diff
@@ -722,13 +722,9 @@ const GovernanceSideBar: FC<IGovernanceSidebarProps> = (props) => {
 				setLoading(false);
 				return;
 			}
-			const contract = new web3.eth.Contract(abi, contractAddress);
-			contract.methods
+			const contract = new Contract(contractAddress, abi, await web3.getSigner());
+			contract
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/index.tsx:726]: need an `await` here

File: src/components/Settings/DemocracyUnlock.tsx [hunk 5]
```diff
@@ -191,17 +191,13 @@ const DemocracyUnlock: FC<IDemocracyUnlockProps> = ({ className, isBalanceUpdate
...
 		setLoadingStatus({ isLoading: true, message: 'Waiting for confirmation' });
 
-		const contract = new web3.eth.Contract(abi, contractAddress);
+		const contract = new Contract(contractAddress, abi, await web3.getSigner());
 
 		// estimate gas.
 		// https://docs.moonbeam.network/builders/interact/eth-libraries/deploy-contract/#interacting-with-the-contract-send-methods
 
-		contract.methods
+		contract
```
Comment (alphainfinitus) [inline src/components/Settings/DemocracyUnlock.tsx:199]: need an `await` here

File: src/components/Settings/DemocracyUnlock.tsx [hunk 6]
```diff
@@ -241,17 +236,13 @@ const DemocracyUnlock: FC<IDemocracyUnlockProps> = ({ className, isBalanceUpdate
...
```
Comment (alphainfinitus) [inline src/components/Settings/DemocracyUnlock.tsx:244]: need an `await` here

File: src/components/Settings/ReferendaUnlock.tsx [hunk 7]
```diff
@@ -293,17 +293,13 @@ const ReferendaUnlock: FC<IReferendaUnlockProps> = ({ className, isBalanceUpdate
...
 		});
 
-		const contract = new web3.eth.Contract(abi, contractAddress);
+		const contract = new Contract(contractAddress, abi, await web3.getSigner());
 
 		// estimate gas.
 		// https://docs.moonbeam.network/builders/interact/eth-libraries/deploy-contract/#interacting-with-the-contract-send-methods
 
-		contract.methods
+		contract
```
Comment (alphainfinitus) [inline src/components/Settings/ReferendaUnlock.tsx:301]: need an `await` here

File: src/components/Settings/ReferendaUnlock.tsx [hunk 8]
```diff
@@ -367,17 +363,13 @@ const ReferendaUnlock: FC<IReferendaUnlockProps> = ({ className, isBalanceUpdate
...
```
Comment (alphainfinitus) [inline src/components/Settings/ReferendaUnlock.tsx:371]: need an `await` here

File: src/ui-components/VotesHistory.tsx [hunk 9]
```diff
@@ -234,13 +234,9 @@ const VotesHistory = ({ className, userProfile, theme, statsArr, setStatsArr, to
 				setRemoveVoteLoading({ ids: [...(removeVoteLoading?.ids || []), postIndex], loading: false });
 				return;
 			}
-			const contract = new web3.eth.Contract(abi, contractAddress);
-			contract.methods
+			const contract = new Contract(contractAddress, abi, await web3.getSigner());
+			contract
```
Comment (alphainfinitus) [inline src/ui-components/VotesHistory.tsx:238]: need an `await` here

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments.

File: package.json [hunk 10]
```diff
@@ -6,6 +6,7 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
+    "analyze": "cross-env ANALYZE=true next build",
```
Comment (alphainfinitus) [inline package.json:9]: please remove the changes related ot the bundle analyzer

**Comment by niklabh** [comment]
when is this getting merged.

**Comment by alphainfinitus** [comment]
Confirm button is always disabled.

**Pull request merged**